File: src/echo-actors/src/main/scala/echo/actor/directory/repository/FeedRepository.scala
package echo.actor.directory.repository

import echo.core.domain.entity.{FeedEntity}
import org.springframework.data.jpa.repository.{JpaRepository, Query}
import org.springframework.data.repository.query.Param

/**
  * @author Maximilian Irro
  */
trait FeedRepository extends JpaRepository[FeedEntity, java.lang.Long] {

    def findOneByEchoId(echoId: String): FeedEntity

    def findAllByUrl(url: String): java.util.List[FeedEntity]

    @Query("SELECT DISTINCT feed FROM FeedEntity feed WHERE feed.podcast.echoId = :podcastExo")
    def findAllByPodcast(@Param("podcastExo") podcastExo: String): java.util.List[FeedEntity]

    @Query("SELECT DISTINCT feed FROM FeedEntity feed WHERE feed.url = :url AND feed.podcast.echoId = :podcastExo")
    def findOneByUrlAndPodcastEchoId(@Param("url") url: String,
                                     @Param("podcastExo") podcastExo: String): FeedEntity

    @Query("SELECT count(feed) FROM FeedEntity feed")
    def countAll(): Long

    @Query("SELECT count(feed) FROM FeedEntity feed WHERE feed.podcast.echoId = :podcastExo")
    def countByPodcast(@Param("podcastExo") podcastExo: String): Long

}

File: src/echo-microservices/app-catalog/src/main/java/echo/microservice/catalog/repository/FeedRepository.java
package echo.microservice.catalog.repository;

import echo.core.domain.entity.FeedEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.List;

/**
 * @author Maximilian Irro
 */
public interface FeedRepository extends JpaRepository<FeedEntity,Long> {

    FeedEntity findOneByExo(String echoId);

    List<FeedEntity> findAllByUrl(String url);

    @Query("SELECT DISTINCT feed FROM FeedEntity feed WHERE feed.podcast.exo = :podcastExo")
    List<FeedEntity> findAllByPodcast(@Param("podcastExo") String podcastExo);

    @Query("SELECT DISTINCT feed FROM FeedEntity feed WHERE feed.url = :url AND feed.podcast.exo = :podcastExo")
    FeedEntity findOneByUrlAndPodcastExo(@Param("url") String url,
                                         @Param("podcastExo") String podcastExo);

    @Query("SELECT count(feed) FROM FeedEntity feed")
    Long countAll();

    @Query("SELECT count(feed) FROM FeedEntity feed WHERE feed.podcast.exo = :podcastExo")
    Long countByPodcast(@Param("podcastExo") String podcastExo);

}
